[DO NOT MERGE] SMPP Connector - Bulk messaging section (#2860)
wso2/docs-ei: en/micro-integrator/docs/references/connectors/smpp-connector/smpp-connector-config.md

       <td>Optional</td>
    </tr>
    <tr>

>        <td>destinationAddress</td>

Reviewer: can't change this. Original implementation carries the word 'distination' by mistake. Can't change it to maintain backward compatibility.

   <description/>
</proxy>       
```
**Note**: For more information on how this works in an actual scenario, see [SMPP Connector Example](smpp-connector-example.md).

## Send bulk SMS message

??? note "sendBulkSMS"
    Used to send SMS messages to multiple destinations.
    <table>
    <tr>
       <th>Parameter Name</th>
       <th>Description</th>
       <th>Required</th>
    </tr>
    <tr>
       <td>serviceType</td>
       <td>

>           Indicates SMS application service. The following generic service_types are defined:

Reviewer 2: ```suggestion
          Indicates the SMS application service used. The following generic service_types are defined:
```

                <td>"USSD"</td>
                <td>Unstructured Supplementary Services Data</td>
             </tr>
          </table>
       </td>
       <td>Optional</td>
    </tr>
    <tr>
       <td>sourceAddressTon</td>
       <td>Type of number for source address.</td>
       <td>Optional</td>
    </tr>
    <tr>
       <td>sourceAddressNpi</td>
       <td>Numbering plan indicator for source address.</td>
       <td>Optional</td>
    </tr>
    <tr>
       <td>sourceAddress</td>
       <td>Source address of the SMS message.</td>
       <td>Yes</td>
    </tr>
    <tr>
      <td>destinationAddresses</td>
      <td>
      Destination addresses can be defined in the following 3 formats.
      <ol>
      <li>Default Values: Default numbering plan and type will be added for all numbers</li>
      <code>
      {"mobileNumbers": ["+94715XXXXXX", "+1434XXXXXX"]}
      </code>
      <li> Custom numbering plan and type for all numbers. The specified plan and type will be set to all numbers.</li>
      <pre>
      {
        "type": "NATIONAL",
        "numberingPlan":"NATIONAL",
        "mobileNumbers": [
            "+94713089759",
            "+189718674"
        ]
      }
      </pre>

>       <li>Each number assigned different properties. Numbers without the properties will be assigned default values (You can set a custom type and numbering plan as default, as mentioned in the previous format). Properties can be set to numbers individually.</li>

Reviewer: Each number **is**

      <pre>
      {
      "mobileNumbers": 
      ["+9471XXXXXX", "+189718674",{ "type": "INTERNATIONAL", "numberingPlan": "NATIONAL", "mobileNumber": "222333" }]
      }
      </pre>
      </ol>
      </td>
      <td>Yes</td>
    </tr>
    <tr>
       <td>message</td>
       <td>Content of the SMS message.</td>
       <td>Yes</td>
    </tr>
    <tr>
       <td>esmClass</td>
       <td>

>           The esmClass parameter is used to indicate special message attributes associated with the short Message(message mode and type).

Reviewer 2: ```suggestion
          The esmClass parameter is used to indicate special message attributes associated with the short Message (message mode and type).
```

             <tr>
                <td>1</td>
                <td>Replace</td>
             </tr>
             <tr>
                <td>2-255</td>
                <td>reserved</td>
             </tr>
             <td>Optional</td>
             <tr>
                <td>alphabet</td>
                <td>
                   Alphabet is used in the data encoding of SMS message. Following alphabets are supported.
                   <ol>
                      <li>ALPHA_DEFAULT</li>
                      <li>ALPHA_8_BIT</li>
                      <li>ALPHA_UCS2</li>
                      <li>ALPHA_RESERVED</li>
                   </ol>
                </td>
                <td>Optional</td>
             </tr>
             <tr>
                <td>isCompressed</td>
                <td>It allows SMS message compression.</td>
                <td>Optional</td>
             </tr>
             <tr>
                <td>messageClass</td>
                <td>
                   <table>
                      <tr>
                         <th>Value</th>
                         <th>Message Class</th>
                      </tr>
                      <tr>
                         <td>CLASS0</td>

>                          <td>Flash messages. Display only not store into the phone</td>

Reviewer 2: ```suggestion
                         <td>Flash messages. These are display only and not stored into the phone</td>
```

                      </tr>
                      <tr>
                         <td>CLASS1</td>
                         <td>ME specific - the SMS is stored in the mobile phone memory</td>
                      </tr>
                      <tr>
                         <td>CLASS2</td>
                         <td>SIM specific - the SMS is stored on the SIM</td>
                      </tr>
                      <tr>
                         <td>CLASS3</td>
                         <td>TE specific - this means the SMS is sent to a computer attached to the receiving mobile phone</td>
                      </tr>
                   </table>
                   Data encoding - defines the encoding scheme of the SMS message. You can find general data coding scheme from [here](https://en.wikipedia.org/wiki/Data_Coding_Scheme) for different combination of alphabet, message class, isCompressed values.
                </td>
                <td>Optional</td>
             </tr>
             <tr>
                <td>submitDefaultMsgId</td>
                <td>

>                    Indicates short message to send from a predefined list of messages stored on SMSC.<br>

Reviewer 2: ```suggestion
                   Indicates a short message to send from a predefined list of messages stored on SMSC.<br>
```

    <alphabet>{$ctx:alphabet}</alphabet>
    <message>{$ctx:message}</message>
    <smscDeliveryReceipt>{$ctx:smscDeliveryReceipt}</smscDeliveryReceipt>
    <messageClass>{$ctx:messageClass}</messageClass>
    <isCompressed>{$ctx:isCompressed}</isCompressed>
    <esmclass>{$ctx:esmclass}</esmclass>
    <protocolid>{$ctx:protocolid}</protocolid>
    <priorityflag>{$ctx:priorityflag}</priorityflag>
    <replaceIfPresentFlag>{$ctx:replaceIfPresentFlag}</replaceIfPresentFlag>
    <submitDefaultMsgId>{$ctx:submitDefaultMsgId}</submitDefaultMsgId>
    <validityPeriod>{$ctx:validityPeriod}</validityPeriod>
    </SMPP.sendBulkSMS>
    ```
    
    **Sample request**
    
    Following is a sample REST/JSON request that can be handled by the sendbulkSMS operation.
    ```json
      {
      "host": "127.0.0.1",
      "port": 2775,
      "systemId": "DAMIEN",
      "password": "neimad",
      "systemType": "UNKNOWN",
      "addressTon": "UNKNOWN",
      "addressNpi": "UNKNOWN",
      "serviceType": "CMT",
      "sourceAddressTon": "NETWORK_SPECIFIC",
      "sourceAddressNpi": "INTERNET",
      "sourceAddress": "16116",
      "destinationAddresses": {
            "type": "ALPHANUMERIC",

>                "numberingPlan": "LAND_MOBILE",

Reviewer: is there a formatting issue here?

>             "mobileNumbers": ["+zad", "+189718674"]

Reviewer: Shall we give a realistic number other than zad
